src/coreclr/jit/codegenriscv64.cpp

```diff
@@ -0,0 +1,7757 @@
+                assert(argOffsetOut <= argOffsetMax); // We can't write beyond the outgoing arg area
+
+                structOffset += moveSize;
+            }
+        }
+    }
+}
+
+//---------------------------------------------------------------------
+// genPutArgReg - generate code for a T_PUTARG_REG node
+//
+// Arguments
+//    tree - the GT_PUTARG_REG node
+//
+// Return value:
+//    None
+//
+void CodeGen::genPutArgReg(GenTreeOp* tree)
+{
+    assert(tree->OperIs(GT_PUTARG_REG));
+
+    var_types targetType = tree->TypeGet();
+    regNumber targetReg  = tree->GetRegNum();
+
+    assert(targetType != TYP_STRUCT);
+
+    GenTree* op1 = tree->gtOp1;
+    genConsumeReg(op1);
+
+    // If child node is not already in the register we need, move it
+    GetEmitter()->emitIns_Mov(ins_Copy(targetType), emitActualTypeSize(targetType), targetReg, op1->GetRegNum(), true);
+    genProduceReg(tree);
+}
+
+//---------------------------------------------------------------------
+// genPutArgSplit - generate code for a GT_PUTARG_SPLIT node
+//
+// Arguments
+//    tree - the GT_PUTARG_SPLIT node
+//
+// Return value:
+//    None
+//
+void CodeGen::genPutArgSplit(GenTreePutArgSplit* treeNode)
+{
+    NYI_RISCV64("genPutArgSplit-----unimplemented/unused on RISCV64 yet----");
+}
+
+//------------------------------------------------------------------------
+// genRangeCheck: generate code for GT_BOUNDS_CHECK node.
+//
+void CodeGen::genRangeCheck(GenTree* oper)
+{
+    noway_assert(oper->OperIs(GT_BOUNDS_CHECK));
+    GenTreeBoundsChk* bndsChk = oper->AsBoundsChk();
+
+    GenTree* arrLen    = bndsChk->GetArrayLength();
+    GenTree* arrIndex  = bndsChk->GetIndex();
+    GenTree* arrRef    = NULL;
+    int      lenOffset = 0;
+
+    GenTree*     src1;
+    GenTree*     src2;
+    regNumber    reg1;
+    regNumber    reg2;
+    emitJumpKind jmpKind = EJ_jmp;
+
+    genConsumeRegs(arrIndex);
+    genConsumeRegs(arrLen);
+
+    emitter*             emit     = GetEmitter();
+    GenTreeIntConCommon* intConst = nullptr;
+    if (arrIndex->isContainedIntOrIImmed())
+    {
+        src1 = arrLen;
+        src2 = arrIndex;
+        reg1 = rsGetRsvdReg();
+        reg2 = src1->GetRegNum();
+
+        intConst    = src2->AsIntConCommon();
+        ssize_t imm = intConst->IconValue();
+        if (imm == INT64_MAX)
+        {
+            emit->emitIns_R_R_I(INS_addi, EA_PTRSIZE, rsGetRsvdReg(), REG_R0, -1);
+            emit->emitIns_R_R_I(INS_srli, EA_PTRSIZE, rsGetRsvdReg(), rsGetRsvdReg(), 1);
+        }
+        else
+        {
+            emit->emitIns_I_la(EA_PTRSIZE, rsGetRsvdReg(), imm);
+        }
+    }
+    else
+    {
+        src1 = arrIndex;
+        src2 = arrLen;
+        reg1 = src1->GetRegNum();
+
+        if (src2->isContainedIntOrIImmed())
+        {
+            reg2        = rsGetRsvdReg();
+            ssize_t imm = src2->AsIntConCommon()->IconValue();
+            emit->emitIns_I_la(EA_PTRSIZE, rsGetRsvdReg(), imm);
```

Comment:
What is the point of marking these immediates as contained if codegen is going to materialize them into a register anyway? Should `GT_BOUNDS_CHECK` just be removed from `IsContainableImmed` instead?

Reply:
I removed `GT_BOUNDS_CHECK` from `IsContainableImmed`.

Reply:
The special cased integer constant codegen in `genRangeCheck` can be removed now -- if they are not marked contained, then codegen will just handle them as usual and you can use their registers directly.

Reply:
Thank you! Fixed!!